compiler/src/steps/import-graph.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { writeFile } from 'fs/promises'
+import { join } from 'path'
+import * as model from '../model/metamodel'
+import { JsonSpec } from '../model/json-spec'
+
+interface ImportTypesGraph {
+  type: model.TypeName
+  imported_by: model.TypeName[]
+}
+
+interface ImportNamespaceGraph {
+  namespace: string
+  imported_by: string[]
+}
+
+export default async function createImportGraph (model: model.Model, jsonSpec: Map<string, JsonSpec>): Promise<model.Model> {
+  const importGraph: Map<model.TypeName, model.TypeDefinition[]> = new Map()
+  for (const current of model.types) {
```

Comment:
It's more complex than that, as we have to traverse every `ValueOf` (extends, generics, property types, etc) to build the graph from `TypeName` values.